src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -234,25 +234,47 @@ def check_cluster_DNS(dns_check_log, filepath_with_timestamp, storage_space_avai
-        if consts.Outbound_Connectivity_Check_Result_String not in outbound_connectivity_check_log:
+        if consts.Outbound_Connectivity_Check_Check_String not in outbound_connectivity_check_log:
             return consts.Diagnostic_Check_Incomplete, storage_space_available
+
+        Outbound_Connectivity_Log_For_Cluster_Connect = outbound_connectivity_check_log.split('  ')[0]
+        # extracting the endpoints for cluster connect feature
+        Cluster_Connect_Enpoint_Url = Outbound_Connectivity_Log_For_Cluster_Connect.split(" : ")[1]
+        # extracting the obo endpoint response code from outbound connectivity check
+        Cluster_Connect_Enpoint_response_code = Outbound_Connectivity_Log_For_Cluster_Connect.split(" : ")[2]
+
+        if(Cluster_Connect_Enpoint_response_code != "000"):
+            if storage_space_available:
+                cluster_connect_outbound_connectivity_check_path = os.path.join(filepath_with_timestamp, consts.Outbound_Network_Connectivity_Check_for_cluster_connect)
+                with open(cluster_connect_outbound_connectivity_check_path, 'w+') as outbound:
+                        outbound.write("Response code " + Cluster_Connect_Enpoint_response_code + "\nOutbound network connectivity check to cluster connect endpoints passed successfully.")
+        else:
+            logger.warning("The outbound network connectivity check has failed for the endpoint - " + Cluster_Connect_Enpoint_Url + "\nThis will affect the \"cluser-connect\" feature. If you are planning to use \"cluster-connect\" functionality , please ensure outbound connectivity to the above endpoint.\n")
+            telemetry.set_exception(exception='Outbound network connectivity check failed for the Cluster Connect endpoint', fault_type=consts.Outbound_Connectivity_Check_Failed_For_Cluster_Connect_Endpoint, summary="Outbound network connectivity check failed for the Cluster Connect endpoint")
+            if storage_space_available:
+                cluster_connect_outbound_connectivity_check_path = os.path.join(filepath_with_timestamp, consts.Outbound_Network_Connectivity_Check_for_cluster_connect)
+                with open(cluster_connect_outbound_connectivity_check_path, 'w+') as outbound:
+                        outbound.write("Response code " + Cluster_Connect_Enpoint_response_code + "\nOutbound connectivity failed for the endpoint:" + Cluster_Connect_Enpoint_Url + " ,this is an optional endpoint needed for cluster-connect feature.")
+
+        Onboarding_Endpoint_outbound_connectivity_response = outbound_connectivity_check_log[-1:-4:-1]
+        Onboarding_Endpoint_outbound_connectivity_response = Onboarding_Endpoint_outbound_connectivity_response[::-1]
+
         # Validating if outbound connectiivty is working or not and displaying proper result
-        if(outbound_connectivity_response != "000"):
+        if(Onboarding_Endpoint_outbound_connectivity_response != "000"):
```

Comment:
I think we should avoid calling "cluster_connect_endpoint" and "onboarding_endpoint" to avoid confusion. May be something like "Cluster_Connect_PreCheck_Endpoint" and "Onboarding_PreCheck_Endpoint"